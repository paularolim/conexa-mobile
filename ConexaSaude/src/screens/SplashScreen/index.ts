export * from './SplashScreen';
export * from './SplashScreenView';
