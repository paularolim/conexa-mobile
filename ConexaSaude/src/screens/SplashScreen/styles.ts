import { Image, View } from 'react-native';
import styled from 'styled-components/native';

export const Container = styled(View)`
  background-color: ${({ theme }) => theme.colors.white};
  padding: 0px 48px;
  flex: 1;
  align-items: center;
  justify-content: center;
`;

export const Logo = styled(Image)`
  max-width: 100%;
  resize-mode: contain;
`;
