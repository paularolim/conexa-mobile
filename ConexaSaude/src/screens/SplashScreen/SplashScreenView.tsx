import logo from '@assets/images/logo.png';

import { Container, Logo } from './styles';

export function SplashScreenView() {
  return (
    <Container>
      <Logo source={logo} />
    </Container>
  );
}
