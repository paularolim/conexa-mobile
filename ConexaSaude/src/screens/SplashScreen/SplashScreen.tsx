import { SplashScreenView } from './SplashScreenView';

export function SplashScreen() {
  return <SplashScreenView />;
}
