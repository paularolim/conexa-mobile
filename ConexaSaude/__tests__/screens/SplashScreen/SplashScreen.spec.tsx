import renderer from 'react-test-renderer';

import { SplashScreen } from '@screens/SplashScreen';

import { Wrapper } from '../../Wrapper';

describe('SplashScreenView', () => {
  it('should render screen correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <SplashScreen />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
