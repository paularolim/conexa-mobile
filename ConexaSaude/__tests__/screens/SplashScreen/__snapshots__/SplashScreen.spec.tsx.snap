// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreenView should render screen correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingLeft": 48,
        "paddingRight": 48,
        "paddingTop": 0,
      },
    ]
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/logo.png",
      }
    }
    style={
      [
        {
          "maxWidth": "100%",
          "resizeMode": "contain",
        },
      ]
    }
  />
</View>
`;
